internal/pkg/template/override/rule.go

```diff
@@ -32,12 +41,21 @@ type nodeUpserter interface {
 
 // Rule is the override rule override package uses.
 type Rule struct {
-	// PathSegment example: "ContainerDefinitions[0].Ulimits.HardLimit"
-	// PathSegment string
-	// Value       *yaml.Node
+	Path  string // example: "ContainerDefinitions[0].Ulimits[-].HardLimit"
+	Value *yaml.Node
 }
 
 func (r Rule) validate() error {
+	if r.Path == "" {
+		return fmt.Errorf("rule path is empty")
+	}
+	pathSegments := strings.Split(r.Path, pathSegmentSeparator)
+	for _, pathSegment := range pathSegments {
+		if !pathSegmentRegexp.MatchString(pathSegment) {
+			return fmt.Errorf(`unrecognized path segment pattern %s. Valid path segment examples are "xyz[0]", "xyz[%s]" or "xyz"`,
```

Comment:
What do you think of:
```suggestion
			return fmt.Errorf(`invalid override path segment '%s': segments must be of the form "array[0]", "map[key]" or "key"`,
```

Reply:
It is not possible for "map[key]"...It would be "foo.bar" instead.

Reply:
Yeah sorry lol 🤦